### ISSUE TITLE
Add support for sorted groups

### ISSUE THREAD

**Issue by fhueske** [stratosphere/stratosphere#168]
Add support for sorted groups (aka secondary sort) for groupBy and coGroup data streams.
In the generic case of custom sort comparators for groupBy this could look for example like:

```
myInput groupBy { _._1 } 
  sortBy { (x,y) => if (x._2 < y._2) 1 else if (x._2 == y._2) 0 else -1 }
  hadoopReduce { doSomething }
```

There should also be shortcuts for (asc, desc) sorts on tuple fields.

**Comment by StephanEwen** [comment]
If we want this to work properly with the current optimizer and runtime, we should use field selectors in the sorting clause. Similarly as we set ordering now not through a comparator, but through fields.

What you suggest would work, I guess, on the iterators in the groups. Should be possible to express that as part of the function even now. Iterator-> toList -> sort (comparator)

Do you think we should still add this as a convenience hook?

**Comment by fhueske** [comment]
You're right. Key selectors are required to make it work with the optimizer and runtime. I didn't consider that.

Sort orders that cannot be expressed with key selectors can still be done within the user code (= without runtime support and additional execution costs). Not sure if a build-in function is necessary. It might hide the additional costs and users might prefer it over the "free" key selector one. In any case it is not a high priority feature.

Do you have an intuition how different sort orders on selected keys could be realized or would you simply use ascending sorting for each selected field?
